src/NServiceBus.Transport.RabbitMQ.TransportTests/ConfigureRabbitMQTransportInfrastructure.cs

```diff
@@ -55,7 +46,38 @@ public Task Cleanup()
 
     void PurgeQueues()
     {
-        var connectionFactory = CreateConnectionFactory();
+        if (connectionStringBuilder == null)
+        {
+            return;
+        }
+
+        var connectionFactory = new ConnectionFactory
+        {
+            AutomaticRecoveryEnabled = true,
+            UseBackgroundThreadsForIO = true
+        };
+
+        object value;
+
+        if (connectionStringBuilder.TryGetValue("username", out value))
+        {
+            connectionFactory.UserName = value?.ToString();
```

Comment:
I took a quick look at the DbConnectionStringBuilder code [here](https://github.com/Microsoft/referencesource/blob/master/System.Data/System/Data/Common/DbConnectionStringBuilder.cs), and it looks like there's no way for value to be null if it successfully retrieved a value. The stored value also always gets stored as a string, so we could just cast instead of call `ToString`.